numpyro/distributions/transforms.py

```diff
@@ -791,6 +792,78 @@ def log_abs_det_jacobian(self, x, y, intermediates=None):
         return -x_abs - 2 * jnp.log1p(jnp.exp(-x_abs))
 
 
+class SimplexToOrderedTransform(Transform):
+    """
+    Transform a simplex into an ordered vector (via difference in Logistic CDF between cutpoints)
+    Used in [1] to induce a prior on latent cutpoints via transforming ordered category probabilities.
+
+    :param anchor_point: Anchor point is a nuisance parameter to improve the identifiability of the transform.
+        For simplicity, we assume it is a scalar value. For more details please refer to Section 2.2 in [1]
+
+    **References:**
+
+    1. *Ordinal Regression Case Study, section 2.2*,
+       M. Betancourt, https://betanalpha.github.io/assets/case_studies/ordinal_regression.html
+
+    """
+
+    domain = constraints.simplex
+    codomain = constraints.ordered_vector
+
+    def __init__(self, anchor_point=0.0):
+        self.anchor_point = anchor_point
+
+    def __call__(self, x):
+        def scan_fn(expit_term, x):
+            y = self.anchor_point - logit(expit_term - x)
+            next_expit_term = expit(self.anchor_point - y)
+            return next_expit_term, y
+
+        # the first term defies the general map formula
+        # hence, the most succinct implementation is to parametrize the changing term
+        expit_init = jnp.ones(x.shape[:-1])
+        # scan must run along the last dimension, so we need to move that dimension to be the leading
+        _, y = lax.scan(scan_fn, expit_init, jnp.moveaxis(x[..., :-1], -1, 0))
```

Comment:
I think `scan` is unnecessary. The transform from cutpoints to probs is
```
x = x - self.anchor_point
s = expit(x)
# I prefer to thinking this way, rather than p0 = 1 - expit(-x0) as in Stan thread
# p0 = s0, p1 = s1 - s0, p2 = s2 - s1,..., pn = 1 - s[n-1]
# add two boundary points 0 and 1
pad_width = [(0, 0)] * (jnp.ndim(s) - 1) + [(1, 1)]
s = jnp.pad(s, pad_width, constant_values=(0, 1))
p = s[..., 1:] - s[..., :-1]
```
and the inverse
```
s = jnp.cumsum(p[..., :-1], axis=-1)
x = logit(s) + self.anchor_point
```

Reply:
Very ellegant! Updated in the latest push